text/0032-minimal-relay.md

```diff
@@ -0,0 +1,196 @@
+# RFC-0032: Minimal Relay
+
+|                 |                                                                               |
+| --------------- | ----------------------------------------------------------------------------- |
+| **Start Date**  | 20 September 2023                                                             |
+| **Description** | Proposal to minimise Relay Chain functionality.                               |
+| **Authors**     | Joe Petrowski, Gavin Wood                                                     |
+
+## Summary
+
+The Relay Chain contains most of the core logic for the Polkadot network. While this was necessary
+prior to the launch of parachains and development of XCM, most of this logic can exist in
+parachains. This is a proposal to migrate several subsystems into system parachains.
+
+## Motivation
+
+Polkadot's scaling approach allows many distinct state machines (known generally as parachains) to
+operate with common guarantees about the validity and security of their state transitions. Polkadot
+provides these common guarantees by executing the state transitions on a strict subset (a backing
+group) of the Relay Chain's validator set.
+
+However, state transitions on the Relay Chain need to be executed by _all_ validators. If any of
+those state transitions can occur on parachains, then the resources of the complement of a single
+backing group could be used to offer more cores. As in, they could be offering more coretime (a.k.a.
```

Comment:
I have seen this argument crop up but I will comment to lay out the counterarguments:
  1. This presumes that the resources spent on including user transactions in blocks _can_ necessarily be spent on adding more cores.
  2. If (1) holds, this presumes that work done on cores will necessarily be more valuable than work done directly on the relay chain

re: (1) Current configurations specify resource requirements for cores (in data / compute) which are discrete and this implies that there is a high likelihood of some leftover resources which are sufficient to execute user transaction but insufficient to add another core. Furthermore, the actual data burden on the network can be estimated by the sizes of PoVs placed onto cores in recent blocks.
re: (2) Synchronous composability is often more valuable than asynchronous

Reply:
At some level, I do think cores could usefully consume otherwise spent on user transactions.  We do not necessarily have a straight line there, but it's still all bandwidth, cpu time, and disk space.  The question is more:  How much resources really?

I think the synchronous vs asynchronous dichotomy winds up being largely illusory here.  We do know some computations parallelize well and some badly, ala npos, but that's not really your dichotomy.  All of web2 is built upon sharded data models aka what you call asynchronous.  It's true that sharded data models require more thought, with this RFC being one example, but they're so dominant in web2 that they're probably not blocking adoption much.

Reply:
I really cannot see any reasonable argument to intentionally put work on an obvious bottleneck point of the network which can be moved into a non-bottleneck point.

Synchronous composability specifically on the Relay-chain, whose entire point under CoreJam is to join async compute results and accumulate state machine transitions, does not sound sensible to me at all.

I've said it before and I'll say it again: The Relay-chain should do one thing and do it well. That means being the head of a secure decentralised multicore computer. Sticking an EVM smart contract in there and/or continuing to allow end-user-application-level transactions and general interaction goes directly against this mantra.